packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -293,21 +326,71 @@ export const ImageInput = ({
     clearInputElement();
   };
 
+  const handlePaste = (event: ClipboardEvent) => {
+    const { files } = event.clipboardData;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
+  const handleDragging = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(true);
+  };
+
+  const handleDragLeave = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(false);
+  };
+
+  const handleDrop = (event: DragEvent) => {
+    handleDragLeave(event);
+    const files = event.dataTransfer?.files;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
   return (
     <Fragment>
-      <InputWrapper>
+      <InputWrapper onPaste={handlePaste}>
         <HiddenInput
           ref={inputRef}
           id={id}
           type="file"
           accept="image/*"
-          onChange={handleChange}
+          onChange={handleInputChange}
           onClick={handleClick}
           disabled={disabled || isLoading}
           aria-invalid={invalid}
           {...props}
         />
-        <StyledLabel isLoading={isLoading} invalid={invalid} htmlFor={id}>
+        <StyledLabel
+          isLoading={isLoading}
+          isDragging={isDragging}
+          invalid={invalid}
+          htmlFor={id}
+          onDragEnter={handleDragging}
+          onDragOver={handleDragging}
```

Comment:
Googling the difference between these two to save you the explanation, but this looks strange

Reply:
https://www.smashingmagazine.com/2018/01/drag-drop-file-uploader-vanilla-js/#drag-and-drop-events

Reply:
I suppose just `dragOver` would be enough 🤔

Reply:
This quote is just 🤯